v2/workloadapi/client_test.go

```diff
@@ -1 +1,400 @@
+	federatedCA := test.NewCA(t)
+	wl := fakeworkloadapi.NewWorkloadAPI(t)
+	defer wl.Stop()
+	c, _ := New(context.TODO(), WithAddr(wl.Addr()))
+	defer c.Close()
+
+	resp := &fakeworkloadapi.X509SVIDResponse{
+		Bundle:           ca.Roots(),
+		SVIDs:            makeX509SVIDs([]string{"spiffe://example.org/foo", "spiffe://example.org/bar"}, ca),
+		FederatedBundles: map[string][]*x509.Certificate{"spiffe://federated.org": federatedCA.Roots()},
+	}
+	wl.SetX509SVIDResponse(resp)
+
+	x509Ctx, err := c.FetchX509Context(context.TODO())
+	assert.Nil(t, err)
+	assert.Len(t, x509Ctx.SVIDs, 2)
+	assert.Len(t, x509Ctx.Bundles.Bundles(), 2)
+}
+
+func TestWatchX509Context(t *testing.T) {
+	ca := test.NewCA(t)
+	federatedCA := test.NewCA(t)
+	wl := fakeworkloadapi.NewWorkloadAPI(t)
+	defer wl.Stop()
+	c, _ := New(context.TODO(), WithAddr(wl.Addr()))
+	defer c.Close()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	tw := newTestWatcher(t)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		_ = c.WatchX509Context(ctx, tw)
+		wg.Done()
+	}()
+
+	// test PermissionDenied
+	tw.WaitForUpdates(1)
+	assert.Len(t, tw.Errors(), 1)
+	assert.Len(t, tw.X509Contexts(), 0)
+
+	// test first update
+	resp := &fakeworkloadapi.X509SVIDResponse{
+		Bundle:           ca.Roots(),
+		SVIDs:            makeX509SVIDs([]string{"spiffe://example.org/foo", "spiffe://example.org/bar"}, ca),
+		FederatedBundles: map[string][]*x509.Certificate{"spiffe://federated.org": federatedCA.Roots()},
+	}
+	wl.SetX509SVIDResponse(resp)
+
+	tw.WaitForUpdates(1)
+
+	assert.Len(t, tw.Errors(), 1)
+	assert.Len(t, tw.X509Contexts(), 1)
+	assert.Len(t, tw.X509Contexts()[0].SVIDs, 2)
+
+	// test second update
+	resp = &fakeworkloadapi.X509SVIDResponse{
+		Bundle:           ca.Roots(),
+		SVIDs:            makeX509SVIDs([]string{"spiffe://example.org/baz"}, ca),
+		FederatedBundles: map[string][]*x509.Certificate{"spiffe://federated.org": federatedCA.Roots()},
+	}
+	wl.SetX509SVIDResponse(resp)
+	tw.WaitForUpdates(1)
+
+	assert.Len(t, tw.Errors(), 1)
+	assert.Len(t, tw.X509Contexts(), 2)
+	assert.Len(t, tw.X509Contexts()[0].SVIDs, 2)
+	assert.Len(t, tw.X509Contexts()[1].SVIDs, 1)
+
+	// test error
+	wl.Stop()
+	tw.WaitForUpdates(1)
+	assert.Len(t, tw.Errors(), 2)
+
+	cancel()
+	wg.Wait()
+}
+
+func TestFetchJWTSVID(t *testing.T) {
+	ca := test.NewCA(t)
+	wl := fakeworkloadapi.NewWorkloadAPI(t)
+	c, _ := New(context.TODO(), WithAddr(wl.Addr()))
+	defer wl.Stop()
+	defer c.Close()
+
+	spiffeID, _ := spiffeid.New("spiffe://example.org", "mytoken")
+	audience := []string{"spiffe://example.org/audience", "spiffe://example.org/extra_audience"}
+	respJWT := makeJWTSVIDResponse(spiffeID.String(), audience, ca)
+	wl.SetJWTSVIDResponse(respJWT)
+
+	params := jwtsvid.Params{
+		Subject:        spiffeID,
+		Audience:       audience[0],
+		ExtraAudiences: []string{"spiffe://example.org/extra_audience"},
+	}
+
+	jwtSvid, err := c.FetchJWTSVID(context.TODO(), params)
+
+	assert.Nil(t, err)
```

Comment:
nit: `require.Nil(t, err)` to stop the test, or the other asserts will panic if this fails